src-tauri/src/platform_specific/linux.rs

```diff
@@ -1,8 +1,78 @@
 // Linux specific code
 
 use regex::Regex;
+use std::env;
 use std::process::Command;
 
+fn get_proton_dir() -> Option<String> {
+    let steam_dir = steamlocate::SteamDir::locate()?;
+    let compat_dir = format!("{}/compatibilitytools.d/", steam_dir.path.display());
+
+    Some(compat_dir)
+}
+
+/// Downloads and installs NS proton
+/// Assumes Steam install
+pub fn install_ns_proton() -> Result<(), thermite::prelude::ThermiteError> {
+    // Get latest NorthstarProton release
+    let latest = thermite::core::latest_release()?;
+
+    let temp_dir = env::temp_dir();
```

Comment:
I really wish I knew about `env::temp_dir` before implementing my own version of temp dir x_x

Time for me to refactor some code again after this xD

Reply:
according to the documentation its always suppose to return PathBuf but life is sometimes more complicated.

Would be worth looking into how its implemented in the case there are some edge cases (e.g. returning empty pathbuf, panicing etc.)

Reply:
okay, just checked up on it.

Its *very* naive.

On systems without a temp dir it just panics, on others it uses assumptions (e.g. /tmp on unix) or calls relevant C functions without any error checking (e.g. GetTempPath2W on windows)

But its offical so /shrug